sdk/storage/Azure.Storage.Internal.Avro/src/AvroParser.cs

```diff
@@ -0,0 +1,472 @@
+        /// </summary>
+        /// <remarks>
+        /// Note that in the Avro spec, byte array length is specified as a long.
+        /// This is fine for Quick Query and Change Feed, but could become a problem
+        /// in the future.
+        /// </remarks>
+        public static async Task<byte[]> ReadFixedBytesAsync(
+            Stream stream,
+            int length,
+            bool async,
+            CancellationToken cancellationToken)
+        {
+            byte[] data = new byte[length];
+            int start = 0;
+            while (length > 0)
+            {
+                int n = async ?
+                    await stream.ReadAsync(data, start, length, cancellationToken).ConfigureAwait(false) :
+                    stream.Read(data, start, length);
+                start += n;
+                length -= n;
+
+                // We hit the end of the stream
+                if (n <= 0)
+                    return data;
+            }
+            return data;
+        }
+
+        /// <summary>
+        /// Reads a single byte from the stream.
+        /// </summary>
+        private static async Task<byte> ReadByteAsync(
+            Stream stream,
+            bool async,
+            CancellationToken cancellationToken)
+        {
+            byte[] bytes = await ReadFixedBytesAsync(stream, 1, async, cancellationToken).ConfigureAwait(false);
+            return bytes[0];
+        }
+
+        /// <summary>
+        /// Internal implementation of ReadLongAsync().
+        /// </summary>
+        private static async Task<long> ReadZigZagLongAsync(
+            Stream stream,
+            bool async,
+            CancellationToken cancellationToken)
+        {
+            byte b = await ReadByteAsync(stream, async, cancellationToken).ConfigureAwait(false);
+            ulong next = b & 0x7FUL;
+            int shift = 7;
+            while ((b & 0x80) != 0)
+            {
+                b = await ReadByteAsync(stream, async, cancellationToken).ConfigureAwait(false);
+                next |= (b & 0x7FUL) << shift;
+                shift += 7;
+            }
+            long value = (long)next;
+            return (-(value & 0x01L)) ^ ((value >> 1) & 0x7fffffffffffffffL);
+        }
+
+        /// <summary>
+        /// Returns null.
+        /// </summary>
+        public static Task<object> ReadNullAsync() => Task.FromResult<object>(null);
+
+        /// <summary>
+        /// Reads a bool from the stream.
+        /// </summary>
+        public static async Task<bool> ReadBoolAsync(
+            Stream stream,
+            bool async,
+            CancellationToken cancellationToken)
+        {
+            byte b = await ReadByteAsync(stream, async, cancellationToken).ConfigureAwait(false);
+
+            if (b == 1)
+                return true;
+            else if (b == 0)
+                return false;
+            else
+                throw new InvalidOperationException("Byte was not a bool.");
```

Comment:
Should we just treat b != 0 as true?

Reply:
Sure?  I don't recall the logic behind this section.

Reply:
It might have been something verbatim from the spec, but given that this implementation is kind of hand wavy around the details I like being permissive.  😄

Reply:
Fixed in next iteration.